### PR TITLE
Migrate DotAiService to data-access lib

### LINKED ISSUE #28953: Separate the Dailog component

### PR DESCRIPTION
**Pull request by nicobytes**
Issue => https://github.com/dotCMS/core/issues/28953

### Proposed Changes
* Migrate `DotAiService` to `@dotcms/data-access`  package
* Migrate models to `@dotcms/dotcms-models`  package
* Fix problems with unit tests

### Checklist
- [x] Tests
- [x] Translations
- [x] Security Implications Contemplated (add notes if applicable)

### Additional Info

This is a first step to migrate `AIImagePromptComponent` to `@dotcms/ui`  package to reusing this component in block-editor and binary field

### Video

Component continues working well

https://github.com/dotCMS/core/assets/7611944/235ffc00-3fca-4a01-b98d-1f13bcac64ee

This PR fixes: #28953

### CODE REVIEW

**Review by hmoreras — APPROVED** [review]

File: core-web/libs/ui/src/lib/services/dot-ai/dot-ai.service.ts
Comment (fmontes) [inline core-web/libs/ui/src/lib/services/dot-ai/dot-ai.service.ts:1]: Shouldn't this be in `data-access`?

**Review by fmontes — CHANGES_REQUESTED** [review]

File: core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/store/ai-content-prompt.store.spec.ts [hunk 1]
```diff
@@ -1,16 +1,15 @@
-import { createServiceFactory, SpectatorService } from '@ngneat/spectator/jest';
+import { createServiceFactory, SpectatorService, SpyObject } from '@ngneat/spectator/jest';
 import { of } from 'rxjs';
 
 import { ComponentStatus } from '@dotcms/dotcms-models';
+import { DotAiService } from '@dotcms/ui';
```
Comment (oidacra) [inline core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/store/ai-content-prompt.store.spec.ts:5]: Move this to `data-access` lib.

File: core-web/libs/ui/src/lib/services/dot-ai/dot-ai.service.ts
Comment (nicobytes) [reply core-web/libs/ui/src/lib/services/dot-ai/dot-ai.service.ts:1]: @fmontes done!

File: core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/store/ai-content-prompt.store.spec.ts (hunk 1 above)
Comment (nicobytes) [reply core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/store/ai-content-prompt.store.spec.ts]: @oidacra done!

**Review by fmontes — APPROVED** [review]

**Review by oidacra — APPROVED** [review]

**Pull request merged**